generator/generator.py

```diff
@@ -1,31 +1,67 @@
-#import numpy
+import numpy as np
 import sys
 import csv
 import random
+import matplotlib.pyplot as plt
+import scipy.stats
 
 def generator(shards, deltaT, N, mean, std):
-    tmp_list = []
     vectors = []
+    tasks_in_deltaT = []
+    requests = []
+    time_marks_in_deltaT = []
 
-    for x in range(shards):
-        for y in range(N):
-            tmp_list.append(random.randint(0, 10))
-        vectors.append(tmp_list)
-        tmp_list = []
-
+    for z in range(shards):        
+        y = np.random.normal(mean, std, N).tolist()
+        y = [round(num, 3) for num in y]
+        vectors.append(y)
+        y = []
 
     vectors_file = open("./generator/vectors.csv","w")
-    for x in vectors:
-        vector_string = str(vectors.index(x)+1) +","+ ','.join(map(str, x)) + "\n"
-        vectors_file.write(vector_string)
-
-
+    for z in vectors:
+       vector_string = str(vectors.index(z)+1) +","+ ','.join(map(str, z)) + "\n"
+       vectors_file.write(vector_string)
+    vectors_file.close()
+            
+    for n in range(N):
```

Comment:
every `for` loop should be exctracted to a separate function and should got an appropriate name

now, we don't know what are these functions responsibities as far as we don't go through them line by line - it takes time and you don't want to start from the lowest level of abstraction when you start an analysis of somebody's else code